src/main/java/com/nutomic/syncthingandroid/activities/ShareActivity.java

```diff
@@ -123,7 +150,34 @@ protected void onCreate(Bundle savedInstanceState) {
             if (files.size() == 1)
                 files.entrySet().iterator().next().setValue(mShareName.getText().toString());
             Folder folder = (Folder) mFoldersSpinner.getSelectedItem();
-            new CopyFilesTask(files, folder).execute();
+            String path = folder.path;
+            if (!TextUtils.isEmpty(mSubDirectoryTextView.getText())) {
+                path += mSubDirectoryTextView.getText();
+            }
```

Comment:
I just spotted another problem, the folder path ends in a / and the subdirectory starts with a /. One of these should be removed.